### PR TITLE
NET-193: Refactor Peer Updates - Firewall Manager

### PR DESCRIPTION
**Pull request by abhishek9686**
## Provide Issue ticket number if applicable/not in title
Decoupled firewall updates from peer updates
## Provide testing steps
refer https://github.com/gravitl/netclient/pull/418
## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [x] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [x] Netmaker is awesome.

### CODE REVIEW

**Review by mattkasun — APPROVED** [review]

**Review by afeiszli — APPROVED** [review]

File: logic/peers.go [hunk 1]
```diff
@@ -425,6 +345,133 @@ func GetPeerUpdateForHost(ctx context.Context, network string, host *models.Host
...
+			if err != nil {
+				logger.Log(1, "no peer host", peer.HostID.String(), err.Error())
+				continue
+			}
+			if node.IsIngressGateway || node.IsEgressGateway {
+				if peer.IsIngressGateway {
+					_, extPeerIDAndAddrs, err := GetExtPeers(&peer)
+					if err == nil {
+						for _, extPeerIdAndAddr := range extPeerIDAndAddrs {
+							extPeerIdAndAddr := extPeerIdAndAddr
```
Comment (TobiaszCudnik) [inline logic/peers.go:391]: I know its moved from elsewhere, but 7-th level of nesting really asks for a separate function...

File: controllers/node.go [hunk 2]
```diff
@@ -461,7 +461,18 @@ func createEgressGateway(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	json.NewEncoder(w).Encode(apiNode)
 	go func() {
-		mq.PublishPeerUpdate()
+		host, err := logic.GetHost(node.HostID.String())
+		if err != nil {
+			logger.Log(0, "failed to get egress host: ", err.Error())
+		}
+		mq.BroadcastAddOrUpdatePeer(host, &node, true)
```
[the remark this answers is not in the document]
Comment (TobiaszCudnik) [reply controllers/node.go:468]: Same happens in `deleteEgressGateway`

**Review by TobiaszCudnik — APPROVED** [review]

File: logic/peers.go (hunk 1 above)
Comment (abhishek9686) [reply logic/peers.go:391]: will be addressed in a diff PR

**Review by TobiaszCudnik — APPROVED** [review]

**Comment by theguy951357** [comment]
cannot get docker client as egress

**Review by theguy951357 — APPROVED** [review]

**Pull request merged**